grpc_opa/authorizer.go

```diff
@@ -240,7 +240,10 @@ func (a *DefaultAuthorizer) Evaluate(ctx context.Context, fullMethod string, opa
 			trace.StringAttribute("out", string(raw)),
 		}, "out")
 	}
-
+	// adding obligations data to context if present
+	if ob,ok := response["obligations"].([]string);ok{
+		ctx = context.WithValue(ctx,"obligations",ob)
```

Comment:
change "obligations" to a private type. See this as an example https://github.com/Infoblox-CTO/atlas.eventing/pull/31/files#diff-c553b059ae0d06bffc1383e72286221924a201e5dc96029dc3b4c9174e46ad93R56

Reply:
The struct{} usage here is a bit unique. An approach I see more often is this
```
type key obligationKey = "obligations"

ctx := context.WithValue(key, ....)
```